src/common/health.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use collection::shards::replica_set::ReplicaState;
+use storage::content_manager::toc::TableOfContent;
+
+pub struct Ready {
+    toc: Arc<TableOfContent>,
+    consensus_sync_instant: Instant,
+}
+
+impl Ready {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self::with_consensus_sync_timeout(toc, Duration::from_secs(10))
+    }
+
+    pub fn with_consensus_sync_timeout(
+        toc: Arc<TableOfContent>,
+        consensus_sync_timeout: Duration,
+    ) -> Self {
+        Self {
+            toc,
+            consensus_sync_instant: Instant::now() + consensus_sync_timeout,
+        }
+    }
+
+    pub async fn ready(&self) -> bool {
+        if Instant::now() < self.consensus_sync_instant {
+            return false;
+        }
```

Comment:
So far, the only reasonable way to handle consensus synchronization on startup that I've come up with, is to return "not ready" for the first `n` seconds after startup.